Elasticsearch storage: support specify the settings `(number_of_shards/number_of_replicas)` for each index individually. (#9914)
apache/skywalking: docs/en/setup/backend/backend-storage.md

the storage provider will use the new username, password, and JKS password to establish the connection and close the old one. If the information exists in the file,
the `user/password` will be overridden.

> ### Each Elasticsearch Index Settings
> You can set and update the index settings `(number_of_shards/number_of_replicas)` by the following configuration items for the index template after rebooting:

Reviewer: ```suggestion
### Index Settings

The following settings control the number of shards and replicas for new and existing index templates. The update only got applied after OAP reboots.
```

```yaml
storage:
  elasticsearch:
    # ......
    indexShardsNumber: ${SW_STORAGE_ES_INDEX_SHARDS_NUMBER:1} 
    indexReplicasNumber: ${SW_STORAGE_ES_INDEX_REPLICAS_NUMBER:1} 
    specificIndexSettings: ${SW_STORAGE_ES_SPECIFIC_INDEX_SETTINGS:""}
    superDatasetIndexShardsFactor: ${SW_STORAGE_ES_SUPER_DATASET_INDEX_SHARDS_FACTOR:5} 

>     superDatasetIndexReplicasNumber: ${SW_STORAGE_ES_SUPER_DATASET_INDEX_REPLICAS_NUMBER:0}

Reviewer: I think as we use merged index policy by default, we could document which indicates affected by which configurations clearly.

    advanced: ${SW_STORAGE_ES_ADVANCED:"{\"index.translog.durability\":\"request\",\"index.translog.sync_interval\":\"5s\"}"}
```

> ### Specify Settings For Each Elasticsearch Index Individually

Reviewer: So, we support other configurations too? Besides shards and replications. Because the original #9849 mentioned these two configurations only.

    advanced: ${SW_STORAGE_ES_ADVANCED:"{\"index.translog.durability\":\"request\",\"index.translog.sync_interval\":\"5s\"}"}
```

#### Specify Settings For Each Elasticsearch Index Individually
You can specify the settings for one or more indexes individually by using `SW_STORAGE_ES_SPECIFIC_INDEX_SETTINGS`.

**NOTE:**
Supported settings:
- number_of_shards
- number_of_replicas

> **NOTE:** If configured, the settings in this configuration have the highest priority and will override the existing 
> generic settings which configured by the others configuration items.

Reviewer: ```suggestion
**NOTE:** These settings have the highest priority and will override the existing 
generic settings mentioned in [index settings doc](#index-settings).
```

Please check whether the new link is correct.

> The settings in `JSON` format and the index name in the config should exclude the `${SW_NAMESPACE}` which is `sw` by default, e.g.

Reviewer: ```suggestion
The settings are in `JSON` format. The index name here is logic entity name, which should exclude the `${SW_NAMESPACE}` which is `sw` by default, e.g.
```

```json
{
  "metrics-all":{
    "number_of_shards":"3",
    "number_of_replicas":"2"
  },
  "segment":{
    "number_of_shards":"6",
    "number_of_replicas":"1"
  }
}
```

> And put it into the following configure item:

Reviewer: ```suggestion
This configuration in the YAML file is like this,
```
